packages/reputation-miner/ReputationMiner.js

```diff
@@ -192,25 +197,35 @@ class ReputationMiner {
   }
 
   /**
-   * When called, adds the entire contents of the current (active) log to its reputation tree. It also builds a Justification Tree as it does so
-   * in case a dispute is called which would require it.
+   * When called, adds the entire contents of the current (active) log to its reputation tree. It also optionally
+   * builds a Justification Tree as it does so in case a dispute is called which would require it.
    * @return {Promise}
    */
-  async addLogContentsToReputationTree(blockNumber = "latest") {
-    if (this.useJsTree) {
-      this.justificationTree = new PatriciaTreeNoHash();
-    } else {
-      const contractFactory = new ethers.ContractFactory(
-        this.patriciaTreeNoHashContractDef.abi,
-        this.patriciaTreeNoHashContractDef.bytecode,
-        this.ganacheWallet
-      );
+  async addLogContentsToReputationTree(blockNumber = "latest", buildJustificationTree = true) {
+    if (buildJustificationTree){
+      if (this.useJsTree) {
```

Comment:
Would be nice to see this pulled out into a separate function, `createJustificationTree("type")` or something which returns either a JS tree, a contract, or the no-op tree.